controllers/tempest_controller.go

```diff
@@ -332,11 +332,11 @@ func (r *TempestReconciler) Reconcile(ctx context.Context, req ctrl.Request) (re
 	//                 It will simplify the logic and duplicite code (Tempest vs Tobiko)
 	if externalWorkflowCounter < len(instance.Spec.Workflow) {
 		if instance.Spec.Workflow[externalWorkflowCounter].NodeSelector != nil {
```

Comment:
Will this condition ever be true after the change? Right now each parameter of the workflow section should be a pointer to easily check whether the value was set or no. Can we maybe keep it this way for now?

Reply:
yes, it can be true having that you can still add node selector to the workflow itself

Reply:
Yes, that's true. Thanks!

I paused here because I wasn't :100:  sure whether we can compare  `maps` and `slices` to `nil` without visiting some unwanted go behaviors. But we should be good here :+1: